job/job.go

```diff
@@ -0,0 +1,202 @@
+package job
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"os/exec"
+	"sync"
+	"syscall"
+
+	"github.com/gopheryan/jobby/streamer"
+)
+
+// Current process state
+type State string
+
+const (
+	// The process is currently running
+	JobStatusRunning State = "RUNNING"
+	// Not necessarily a 'success' (see exit code for that)
+	// This state signals that the process is no longer running
+	JobstatusComplete State = "COMPLETE"
+	// Different from 'COMPLETE' in that this state
+	// means that the user deliberately stopped the job
+	JobStatusStopped State = "STOPPED"
+)
+
+func newState(processExited, userKilled bool, e *exec.ExitError) State {
+	if processExited {
+		// ExitCode() returns -1 if the process hasn't exited or was terminated by a signal
+		// Well we know the process exited
+		if e != nil && e.ExitCode() == -1 {
+			status, ok := e.ProcessState.Sys().(syscall.WaitStatus)
+			// We know the process has exited, and the 'userKilled' flag *may*
+			// be set, but there's a chance that the process exited just as Stop()
+			// was called. Validate that this process was killed by a kill signal
+			if ok && status.Signal() == syscall.SIGKILL && userKilled {
+				// The process has exited, it was killed by a kill signal,
+				// and the job has a 'userKilled' flag indicating that the caller
+				// asked us to stop a running job.
+				return JobStatusStopped
+			}
+		}
+		return JobstatusComplete
+	}
+	return JobStatusRunning
+}
+
+type Status struct {
+	CurrentState State
+	ReturnCode   *int
+}
+
+type JobArgs struct {
+	Command    string
+	Args       []string
+	StdoutPath string
+	StderrPath string
+}
+
+type Job struct {
+	jobLock       sync.Mutex
+	cmd           exec.Cmd
+	processExited bool
+	processDone   chan struct{}
+	exitErr       *exec.ExitError
+	userKilled    bool
+
+	stdoutPath string
+	stderrPath string
+}
+
+func logFileClose(f *os.File) {
+	if err := f.Close(); err != nil {
+		if !errors.Is(err, os.ErrInvalid) {
+			slog.Error("Failed to close file", "error", err)
+		}
+	}
```

Comment:
If `ErrInvalid` is only returned when the file is nil perhaps return early in that case and drop the special error handling?
```suggestion
	if f == nil {
		return
	}
	
	if err := f.Close(); err != nil {
		slog.Error("Failed to close file", "error", err)
	}
```